### PR TITLE
[event monitor][CWS] reintroduce snapshot event playing

### PR DESCRIPTION
**Pull request by YoannGh**
This PR delays the startup of the event monitor eventstream to make it possible to match process data collected by the probe snapshot against CWS rules without any concurrency issue.

Currently, CWS rules are loaded upon starting the CWS consumer, which happens after the runtime security probe goroutine started processing events, hence matching snapshotted process data from another goroutine requires locking the runtime security probe's process data structures which isn't ideal.

To avoid requiring locking, this PR refactors the startup code of the event monitor module the following way:

- Setup the runtime security probe:
  - Start the eBPF manager (create eBPF maps and load eBPF programs)
  - Apply default kernel filters
  - Attach the probes for the default event types common to all consumers + the probe required to populate eBPF maps during the snapshot phase   
- Do the probe snapshot to populate the runtime security probe and eBPF maps process data
- Start the event monitor's consumers, the CWS consumer will load its policies, updating the activated probes and kernel filters in the process
- Start the runtime security probe: 
  - Play the snapshotted process data, potentially triggering CWS rules in the process
  - then start the event stream

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/eventmonitor/eventmonitor.go [hunk 1]
```diff
@@ -139,16 +139,18 @@ func (m *EventMonitor) Start() error {
...
-	}
-
 	// start event consumers
 	for _, em := range m.eventConsumers {
 		if err := em.Start(); err != nil {
 			log.Errorf("unable to start %s event consumer: %v", em.ID(), err)
 		}
 	}
+	// Apply rules to the snapshotted data
+	m.Probe.PlaySnapshot()
```
Comment (safchain) [inline pkg/eventmonitor/eventmonitor.go:149]: If we can I would call the PlaySnapshot function directly from the Start function

Comment (safchain) [inline pkg/eventmonitor/eventmonitor.go:149]: If we can I would call the PlaySnapshot function directly from the Start function

**Review by safchain — APPROVED** [review]
A tiny comment otherwise LGTM

**Review by brycekahle — APPROVED** [review]
shallow review on `pkg/eventmonitor/eventmonitor.go` only.

**Pull request merged**